app/components/Analytics.tsx

```diff
@@ -106,6 +106,24 @@ const Analytics: React.FC = ({ children }: Props) => {
     });
   }, []);
 
+  // Umami
+  React.useEffect(() => {
+    (env.analytics as PublicEnv["analytics"]).forEach((integration) => {
+      if (integration.service !== IntegrationService.Umami) {
+        return;
+      }
+
+      const script = document.createElement("script");
+      script.defer = true;
+      script.src = `${integration.settings?.instanceUrl}/${integration.settings?.scriptName}`;
```

Comment:
This is not acceptable as it would allow injecting any script at all, as far as I can tell it is always hosted at `/script.js` ?

Reply:
Different, umami.js was specified in v1, and then the official default value given in v2 was script.js, and at the same time, self-built users were allowed to set the script name to avoid being blocked by AD advertising plug-ins.

![image](https://github.com/user-attachments/assets/d80cc61f-d419-437c-9634-56f6dea2220a)

You can refer to the official installation documentation of Umami. There is nothing to specify the script name. First of all, this is something that administrators have permission to do. 
This situation is only for self-built Umami, and the time value is fixed when using Umami Cloud.

[https://umami.is/docs/environment-variables](https://umami.is/docs/environment-variables)

Reply:
That's a shame, that does mean that this integration won't be possible to support in cloud then - you might as well change this so that `instanceUrl` is just the entire path to the script then? There isn't really a benefit to storing the parts separately and concatenating

Reply:
According to the original development logic, I don't want to break the `instanceUrl` format and cause more trouble.

Secondly, the `instanceUrl ` should also be added to the SCP permission list, I think, to avoid the trouble of dealing with the `instanceUrl ` modification if it is standardized at a later stage.